### PR TITLE
Form: Reduxify FormCountrySelect component

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request is a follow-up of https://github.com/Automattic/wp-calypso/pull/25304, and relates to https://github.com/Automattic/wp-calypso/issues/18095. It seeks to update the `FormCountrySelect` component (and any component using it) in order to accept a list of countries coming from the Redux global state tree rather than from the old [`CountriesList` data store](https://github.com/Automattic/wp-calypso/tree/2efbed6e01fd3bac3ae8fbf1022ecacb96328f92/client/lib/countries-list) (which it removes in the process).

This pull request touches way too many files, but I'm not sure how to split it given how many components depend on `FormCountrySelect` - as you can see in the following dependency graph I made with [draw.io](https://www.draw.io/) (available from 1dfe1-pb). The 🔗 sign is used to denote a component that is connected to the global state:

screenshot

##### Devdocs

screenshot

##### Two-Step Authentication

screenshot

##### Account Recovery

screenshot

##### Domain Contact Information

screenshot

##### G Suite Account Information

screenshot

##### Ebanx Payment Method

screenshot

##### Redirect Payment Method

TBD.

##### WooCommerce Customer Address

TBD.

#### Testing instructions

1. Run `git checkout update/form-country-select` and start your server, or open a [live branch](https://calypso.live/?branch=update/form-country-select)

##### Devdocs

2. Open the [`Devdocs` example page](http://calypso.localhost:3000/devdocs/design/form-fields)
3. Asserts that the `Form Country Select` select box displays correctly
4. Asserts that the `Country Code` select box under `Form Phone Input` displays correctly 
5. Asserts that the `Form Media Phone Input` field works correctly

##### Two-Step Authentication

2. Open the [`Two-Step Authentication` page](http://calypso.localhost:3000/me/security/two-step)
3. Click the `Get Started` button
4. Asserts that the `Country Code` select box displays correctly 
5. Assert that you can enable two-steps authentication via SMS

##### Account Recovery

2. Navigate to the [`Account Recovery` page](http://calypso.localhost:3000/me/security/account-recovery)
3. Click the `Add` button in front of `Recovery SMS Number`
4. Asserts that the `Country Code` select box displays correctly
5. Assert that you can enable account recovery via SMS

##### Domain Contact Information

2. Open the [`Add Domain` page](http://calypso.localhost:3000/domains/add)
3. Select a domain name, and proceed to checkout
4. Assert that the `Phone` field in the `Domain Contact Information` section displays correctly
5. Assert that the `Country` select box below displays correctly
6. Assert that you can purchase a domain name

##### G Suite Account Information

2. Open the [`Domains` page](http://calypso.localhost:3000/domains)
3. Click a registered domain to access the `Domain Settings` page
4. Click the `Email` navigation action
5. Click the `Add G Suite` button
6. Fill in the `Add Email Addresses` field, and click the `Continue` button to proceed to checkout
7. Assert that the `Country` field in the `G Suite Account Information` section displays correctly
8. Assert that you can purchase a G Suite subscription

##### Ebanx Payment Method

2. Set the currency of your account to `BRL` from the `Store Admin` page
3. Proceed to checkout, e.g. by purchasing a plan
4. Select `Brazil` in the `Country` select box
5. Assert that additional fields are displayed
6. Assert that you can perform the purchase successfully

##### Redirect Payment Method

TBD.

##### WooCommerce Customer Address

TBD.

You can also run unit tests with:

```
npm run test-client client/blocks/credit-card-form
npm run test-client client/components/credit-card-form-fields
npm run test-client client/components/domains/contact-details-form-fields
npm run test-client client/components/forms/form-country-select
npm run test-client client/components/forms/form-phone-input
npm run test-client client/components/payment-country-select
npm run test-client client/my-sites/checkout/checkout
npm run test-client client/my-sites/domains/components/form
```

#### Additional notes

This pull request is preliminary work for https://github.com/Automattic/wp-calypso/pull/25166.

#### Reviews

- [ ] Code
- [ ] Product
- [ ] Tests

### CODE REVIEW

**Comment by stephanethomas** [comment]
This pull request is blocked until https://github.com/Automattic/wp-calypso/pull/25304 is merged.

**Comment by cbauerman** [comment]
Product looks good, I tested everything as described except making sure that the gsuite purchase went through.

Code also looks good to me, even though there is a lot it is nice that it repeats itself frequently!

File: client/blocks/credit-card-form/test/index.js [hunk 1]
```diff
@@ -13,15 +13,19 @@ import React from 'react';
...
  * Internal dependencies
  */
-import CreditCardForm from '../';
+import { CreditCardForm } from '../';
 
 jest.mock( 'i18n-calypso', () => ( {
 	localize: x => x,
 } ) );
 
+// Gets rid of warnings such as 'UnhandledPromiseRejectionWarning: Error: No available storage method found.'
```
Comment (Tug) [inline client/blocks/credit-card-form/test/index.js]: This look like a commit message rather. I think a better comment would be to indicate which dependency causes this warning

File: client/components/forms/form-phone-input/index.jsx [hunk 2]
```diff
@@ -89,8 +89,8 @@ export class FormPhoneInput extends React.Component {
 	}
 
 	getCountryData() {
-		// TODO: move this to country-list or CountrySelect
-		return find( this.props.countriesList.get(), {
+		// TODO: move this to country-list or FormCountrySelect
```
Comment (Tug) [inline client/components/forms/form-phone-input/index.jsx]: I don't think this TODO is needed with your refactor

File: client/my-sites/domains/components/form/country-select.jsx [hunk 3]
```diff
@@ -24,8 +23,6 @@ import FormSelect from 'components/forms/form-select';
 const CountrySelect = createReactClass( {
 	displayName: 'CountrySelect',
 
-	mixins: [ observe( 'countriesList' ) ],
```
Comment (Tug) [inline client/my-sites/domains/components/form/country-select.jsx]: 🎉

**Comment by Tug** [comment]
> Asserts that the Form Media Phone Input field works correctly
I was unable to enter a phone number when trying this component. Clicking on the text field would open the selector each time

**Comment by stephanethomas** [comment]
> I was unable to enter a phone number when trying this component. Clicking on the text field would open the selector each time

That particular issue already exists in `master`, and as far I can see only happens in Devdocs. There is indeed a zone that is as large as the longest item in the select box where any click will open the select box.

**Comment by Tug** [comment]
I added a quick fix and re-tested everything.
This looks ready to me 👍

**Pull request merged**